src/openads/application/service/RefreshPositionUseCase.js

```diff
@@ -0,0 +1,43 @@
+import PositionNotFoundException from '../../domain/position/PositionNotFoundException'
+import {POSITION_NOT_VISIBLE} from '../../domain/position/positionStatus'
+import PositionNotVisibleException from '../../domain/position/PositionNotVisibleException'
+
+export default class RefreshPositionUseCase {
+  /**
+   * Update a Position with given changes and refresh his Ad
+   * @param {PositionRepository} positionRepository
+   */
+  constructor ({positionRepository}) {
+    this._positionRepository = positionRepository
+  }
+  /**
+   * Update a Position with given changes and refresh his Ad
+   * @param {string} id
+   * @param {object} position
+   * @param {string} position.name
+   * @param {string} position.placement
+   * @param {string} position.segmentation
+   * @param {Array} position.sizes
+   * @returns {Promise<Position>}
+   */
+  refreshPosition ({id, position}) {
+    return this._positionRepository.find({id})
+      .then(optionalPosition => this._resolveOptionalPosition({optionalPosition, id}))
+      .then(this._filterPositionVisible)
```

Comment:
![](https://media.giphy.com/media/sHytgXD2vlNVC/giphy.gif)

Reply:
uhmmmm.... but it forces you to use a non name-parametrized function ;)

Reply:
right, named parameters is a problem for functional programming as far as I know...at least the implementation of named parameters of ES6 (tricky)